### ISSUE TITLE
stream rows api blocks the event loop

### ISSUE THREAD

**Issue by dynamikus** [snowflakedb/snowflake-connector-nodejs#593]
Please answer these questions before submitting your issue. 
In order to accurately debug the issue this information is required. Thanks!

1. What version of NodeJS driver are you using?
1.6.23
   
2. What operating system and processor architecture are you using?
Windows
   
3. What version of NodeJS are you using?
(`node --version` and `npm --version`)
20.2.0
4. What are the component versions in the environment (`npm list`)?

5.Server version:* E.g. 1.90.1
7.24.2
6. What did you do?

I am using the stream rows api, to stream the results back. I have a modest data set of 100K.  While data is getting downloaded no other process in the event loop are getting processed. I would assume the culprit is[ process.nextTick()](https://github.com/snowflakedb/snowflake-connector-nodejs/blob/master/lib/connection/result/row_stream.js#L174)

 This is because process.nextTick() is not technically part of the event loop. Instead, the nextTickQueue will be processed after the current operation is completed, regardless of the current phase of the event loop. Basically an other task that is part of the event loop will get deffered till snowflake is done getting the data.

7. What did you expect to see?

I expect streaming data should not block the event loop, so it can give a chance for other processes to run.

**Comment by sfc-gh-dszmolka** [comment]
hello and thank you for submitting this issue. it is quite astonishing to me, because looking at the git blame, this code part has been there forever (but at least since it was originally uploaded to Github, 5 years ago).

of course it doesn't mean it has to be like this going forward, so we're going to take a look how the row stream could be reimplemented to play better along with the eventloop.

**Comment by dynamikus** [comment]
Yeah same here, I have been using this sdk for 4+ years now, and never come across a scenario to process something else while waiting for the data to load till last week. 

Anyway thank you for looking into this, let me know if you want a code example to replicate it.

**Comment by sfc-gh-dszmolka** [comment]
hey folks, if any of you is still watching. This issue wasn't so high priority on the list, but the driver team finally got here and they trying to reproduce the issue with 
```js
import * as snowflake from 'snowflake-sdk';
snowflake.configure({
  logLevel: 'OFF',
  // useEnvProxy: false,
});

const connecOption: snowflake.ConnectionOptions = {
  account: process.env.SNOWFLAKE_GCP_TEST_ACCOUNT || 'testaccount',
  username: process.env.SNOWFLAKE_TEST_USER || 'testuser',
  password: process.env.SNOWFLAKE_GCP_TEST_PASSWORD || 'testpassword',
  database: process.env.SNOWFLAKE_TEST_DATABASE || 'testdatabase',
  schema: process.env.SNOWFLAKE_TEST_SCHEMA || 'testschema',
  warehouse: process.env.SNOWFLAKE_TEST_WAREHOUSE || 'testwarehouse',

  // role: 'SYSADMIN',
  // proxyHost: 'localhost',
  // proxyPort: 3128,
  // proxyProtocol: 'http',
  // useProxyForOCSP: true,
  // noProxy: "*.snowflakecomputing.com",
};

// @ts-ignore

const columns = (num) => {
  let string = '';
  for (let i = 0; i < num; i++) {
    string += `randstr(${i + 1},random()) ${i !== num - 1 ? ',' : ''} `;
  }
  return string;
};

async function run1() {
  const connection = snowflake.createConnection(connecOption);
  await connection.connectAsync((err) => {
    if (err) {
      console.error('Unable to connect: ' + err.message);
    } else {
      console.log('Successfully connected to Snowflake.');
    }
  });

  const executeQuery = (options) =>
    new Promise((resolve, reject) => {
      connection.execute({
        ...options,
        // @ts-ignore
        complete: (err, statement, rows) => {
          if (err) {
            reject(err);
          } else {
            resolve({ statement, rows });
          }
        },
      });
    });
  let count = 0;
  setInterval(() => {
    console.log('Other event is running , count:', count++);
  }, 1000);

  console.time('executeQuery');
  const Result = await executeQuery({
    sqlText: `select ${columns(100)} from table(generator(rowcount=>${100000}))`,
  });
  console.timeEnd('executeQuery');
}

```

and outcome was that executeQuery takes 42.531s and there are 41 occurrences of "Other event is running", which indicates that the event loop is not stuck and is iterating properly.

Is there someone perhaps who still sees issues around this with the recent (2.1.0+) driver versions and would be able to give us a reproduction ? Thank you in advance !

**Comment by sfc-gh-dszmolka** [comment]
closing this ticket due to inactivity. please if it still persist for anyone, on the latest (time of writing: 2.2.0) driver version, open a new one with the reproduction. Thank you !
